app/scripts/services/ahjo.meeting.srv.js

```diff
@@ -59,193 +63,46 @@ angular.module('dashboard')
             return deferred.promise;
         };
 
-        self.getDecisions = function (topicGuid) {
-            var deferred = $q.defer();
+         /*
+         * @name dashboard.utils.getDecisions
+         * @description Returned promise will resolve with response data
+         * or reject with { error: } object containing an HTTP or REST error code or -1 for a badly formatted response.
+         * @param {string} Meeting guid
+         * @param {string} Topic guid
+         * @returns {promise} Promise object
+         */
+        self.getDecisions = function (aMeetingGuid, aTopicGuid) {
+            var def = $q.defer();
             $timeout(function () {
-                deferred.notify({});
+                def.notify({});
+
+                $http({
+                    method: 'GET',
+                    cache: false,
+                    url: ENV.AhjoApi_Decisions.replace(':meetingguid', aMeetingGuid).replace(':topicguid', aTopicGuid)
+                }).then(function (resp) {
+
+                    var errCode = Utils.processAhjoError(resp, false);
+                    if (errCode) {
+                        $log.error("AhjoMeetingSrv.getDecisions: REST error", errCode);
+                        rejectError(errCode);
+                    }
```

Comment:
It seems that execution continues to 'objects' part even 'errorCode' is found. Maybe 'reject' works as 'return'? Or is there need for 'else'?